src/test/scala/akka/contrib/datareplication/ORSetSpec.scala

```diff
@@ -308,19 +309,34 @@ class ORSetSpec extends WordSpec with Matchers {
       val a2 = a.remove(node1, "Z")
       // replicate b to node1, now node1 has node2's 'Z'
       val a3 = a2.merge(b)
-      a3.value should be(Set("Z"))
+      a3.elements should be(Set("Z"))
       // Remove node2's 'Z'
       val b2 = b.remove(node2, "Z")
       // Replicate c to node2, now node2 has node1's old 'Z'
       val b3 = b2.merge(c)
-      b3.value should be(Set("Z"))
+      b3.elements should be(Set("Z"))
       // Merge everytyhing
-      a3.merge(c).merge(b3).value should be(Set.empty)
-      a3.merge(b3).merge(c).value should be(Set.empty)
-      c.merge(b3).merge(a3).value should be(Set.empty)
-      c.merge(a3).merge(b3).value should be(Set.empty)
-      b3.merge(c).merge(a3).value should be(Set.empty)
-      b3.merge(a3).merge(c).value should be(Set.empty)
+      a3.merge(c).merge(b3).elements should be(Set.empty)
+      a3.merge(b3).merge(c).elements should be(Set.empty)
+      c.merge(b3).merge(a3).elements should be(Set.empty)
+      c.merge(a3).merge(b3).elements should be(Set.empty)
+      b3.merge(c).merge(a3).elements should be(Set.empty)
+      b3.merge(a3).merge(c).elements should be(Set.empty)
+    }
+
+    "have unapply extractor" in {
+      val s1 = ORSet.empty.add(node1, "a").add(node2, "b")
+      val s2: ORSet[String] = s1
+      val ORSet(elements1) = s1 // `unapply[A](s: ORSet[A])` is used here
+      val elements2: Set[String] = elements1
+      Changed("key", s1) match {
+        case Changed("key", ORSet(elements3)) => // `unapply(a: ReplicatedData)` is used here
+          // if `unapply(a: ReplicatedData)` isn't defined the next line doesn't compile:
+          //   type mismatch; found : scala.collection.immutable.Set[A] where type A required: Set[Any] Note: A <: Any,
+          //   but trait Set is invariant in type A. You may wish to investigate a wildcard type such as _ <: Any. (SLS 3.2.10)
+          val elements4: Set[Any] = elements3
```

Comment:
@hseeberger @huntc I have improved the support for `unapply` and it works as I like now. Not much to review, but there is one interesting thing that I don't fully understand. I had to add overloads for unapply for `ORSet` and `ORMultiMap`. See comments in the code above. I think it has to do with that `Set` is invariant.